### PR TITLE
Completed list2.py

### PR DESCRIPTION
**Pull request by dsbarnes**

### CODE REVIEW

**Comment by astephens91** [comment]
Just a quick note!  It's well done!

Your code works perfectly in Python 2. But I'm getting the habit of commenting on Python 3 changes since Python 3 will be the standard come January. To that end, for your "front_x" function in list1.py, in Python3, filter() will return an iterable so you can't call the sort method on it. Sort doesn't work on iterables. That said, the fix is to either wrap the filter method object in a list() OR use sorted() instead of sort()

Tldr; Your code works fine in Python2 but not Python3 and I included a tip to make it work in Python3 since it'll soon be the standard.
